testsuites/syncgateway/functional/tests/test_cbs_collections.py

```diff
@@ -326,6 +328,93 @@ def test_restricted_collection(scopes_collections_tests_fixture):
+    scope_user_doc_prefix = "scope_user_doc" + random_suffix
+    collection_user_doc_prefix = "collection_user_doc" + random_suffix
+
+    # Upload docs to both collections as scope_user
+    add_docs_result = sg_client.add_docs(sg_url, db, 3, scope_user_doc_prefix, auth=scope_user_auth, channels=[second_collection], scope=scope, collection=second_collection)
+    doc_ids_second_collection = ([doc["id"] for doc in add_docs_result])
+    add_docs_result = sg_client.add_docs(sg_url, db, 3, scope_user_doc_prefix, auth=scope_user_auth, channels=[collection], scope=scope, collection=collection)
+    doc_ids_collection = [doc["id"] for doc in add_docs_result]
+
+    # Attempt to upload docs to both collections as collection_user
+    add_docs_result = sg_client.add_docs(sg_url, db, 3, collection_user_doc_prefix, auth=collection_user_auth, channels=[collection], scope=scope, collection=collection)
+    doc_ids_collection.extend([doc["id"] for doc in add_docs_result])
+    with pytest.raises(HTTPError) as e:
+        sg_client.add_docs(sg_url, db, 3, collection_user_doc_prefix, auth=collection_user_auth, channels=[second_collection], scope=scope, collection=second_collection)
+    assert("403" in str(e)), "User without access to collection should generate 403 HTTPError when trying to upload documents to that collection. Instead got: \n" + str(e)
+
+    # 4. Check document retrieval via all_docs and document ID is restricted to correct access level
+    # Get all docs as scope_user
+    scope_user_collection_docs = [row["id"] for row in sg_client.get_all_docs(sg_url, db, auth=scope_user_auth, scope=scope, collection=collection)["rows"]]
+    scope_user_second_collection_docs = [row["id"] for row in sg_client.get_all_docs(sg_url, db, auth=scope_user_auth, scope=scope, collection=second_collection)["rows"]]
+    assert(len(scope_user_collection_docs) == len(doc_ids_collection)), f"_all_docs endpoint returned wrong number of docs for user with access to {collection}. Expected {len(doc_ids_collection)}, got {len(scope_user_collection_docs)}"
+    assert(len(scope_user_second_collection_docs) == len(doc_ids_second_collection)), f"_all_docs endpoint returned wrong number of docs for user with access to {second_collection}. Expected {len(doc_ids_second_collection)}, got {len(scope_user_second_collection_docs)}"
+
+    # Get all docs as collection_user
+    collection_user_collection_docs = [row["id"] for row in sg_client.get_all_docs(sg_url, db, auth=collection_user_auth, scope=scope, collection=collection)["rows"]]
+    collection_user_second_collection_docs = [row["id"] for row in sg_client.get_all_docs(sg_url, db, auth=collection_user_auth, scope=scope, collection=second_collection)["rows"]]
+    assert(len(collection_user_collection_docs) == len(doc_ids_collection)), f"_all_docs endpoint returned wrong number of docs for user with access to {collection}. Expected {len(doc_ids_collection)}, got {len(collection_user_collection_docs)}"
```

Comment:
These two lines are under the comment "Get all docs as collection_user", would it be possible to separate them from that comment? They are doing validation, which is a different thing